Objects/tupleobject.c

```diff
@@ -11,6 +11,24 @@ class tuple "PyTupleObject *" "&PyTuple_Type"
 
 #include "clinic/tupleobject.c.h"
 
+/* The objects representing an empty tuple.
+   Empty tuples are used all over the place and applications may
+   rely on the fact that an empty tuple is a singleton. */
+
+static struct {
+    PyGC_Head gc_head;
+    PyTupleObject obj;
+} _PyTuple_EmptyStruct = {
+    .gc_head = { .gc = {
+        .gc_refs = (((size_t)_PyGC_REFS_UNTRACKED) << _PyGC_REFS_SHIFT)
```

Comment:
Frankly this is ugly :-( Can't you just allocate `_PyTuple_Empty` in `PyTuple_Init`?

Reply:
Can you at least try to factor out the `gc_head` initialization thing? For example by defining a macro? Leaking private GC management details in `tupleobject.c` does not look desirable at all.

Reply:
I'll try to do something with this.

Reply:
I'd be much more inclined to figure out where super early on in Py_Initialize we can allocate the empty PyTuple using the old PyTuple_New(0) code path.  I don't like being inconsistent about where pointers to PyObjects come from.  All tuples should be Py_DECREF'ed to 0 during interpreter shutdown but code attempting to free this one would fail in a bad way as it isn't on the heap.